lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -984,31 +1041,122 @@ def default_ami_for_region(region, criteria = {})
+                 begin
+                   if machine_spec.reference['winrm_encrypted_password']
+                     decoded = Base64.decode64(machine_spec.reference['winrm_encrypted_password'])
+                   else
+                     encrypted_admin_password = instance.password_data.password_data
+                     if encrypted_admin_password.nil? || encrypted_admin_password.empty?
+                       raise "You did not specify winrm_password in the machine options and no encrytpted password could be fetched from the instance"
+                     end
+                     machine_spec.reference['winrm_encrypted_password']||=encrypted_admin_password
+                     # ^^ saves encrypted password to the machine_spec
+                     decoded = Base64.decode64(encrypted_admin_password)
+                   end
+                   # decrypt so we can utilize
+                   private_key = OpenSSL::PKey::RSA.new(get_private_key(instance.key_name))
+                   private_key.private_decrypt decoded
+                 end
+
+      disable_sspi =  machine_spec.reference['winrm_disable_sspi'] ||
+                      machine_options[:winrm_disable_sspi] ||
+                      false # default to Negotiate
+      basic_auth_only = machine_spec.reference['winrm_basic_auth_only'] ||
+                        machine_options[:winrm_basic_auth_only] ||
+                        false # disallow Basic auth by default
+      no_ssl_peer_verification = machine_spec.reference['winrm_no_ssl_peer_verification'] ||
+                                 machine_options[:winrm_no_ssl_peer_verification] ||
+                                 false #disallow MITM potential by default
 
       winrm_options = {
-        :user => machine_spec.reference['winrm_username'] || 'Administrator',
-        :pass => password,
-        :disable_sspi => true,
-        :basic_auth_only => true
+        user: username,
+        pass: password,
+        disable_sspi: disable_sspi,
+        basic_auth_only: basic_auth_only,
+        no_ssl_peer_verification: no_ssl_peer_verification,
       }
 
+      if no_ssl_peer_verification or type != :ssl
+        # =>  we won't verify certs at all
+        Chef::Log.info "No SSL or no peer verification"
+      elsif machine_spec.reference['winrm_ssl_thumbprint']
+        # we have stored the cert
+        Chef::Log.info "Using stored fingerprint"
+      else
+        # we need to retrieve the cert and verify it by connecting just to
+        # retrieve the ssl certificate and compare it to what we see in the
+        # console logs
+        instance.console_output.data.output
+        # again this seem to need to be run twice, to ensure
+        encoded_output = instance.console_output.data.output
+        console_lines = Base64.decode64(encoded_output).lines
+        fp_context = OpenSSL::SSL::SSLContext.new
+        tcp_connection = TCPSocket.new(instance.private_ip_address, port)
```

Comment:
Hey @hh - why are you using the private_ip_address here? When I try to connect to an `https` connection using this PR I get the following error

https://gist.github.com/tyler-ball/73996b85e155ab76cfa2280ecac6c71d